### PR TITLE
Track and skip failure causing values

### PR DESCRIPTION
**Pull request by AdityaVallabh**

### CODE REVIEW

File: meqa/mqgo/main.go [hunk 1]
```diff
@@ -311,12 +314,37 @@ func runMeqa(meqaPath, swaggerFile, testPlanFile, resultPath,
...
+		fuzzMode = 1
+	case "datatype":
+		fuzzMode = 2
+	case "negative":
+		fuzzMode = 3
+	default:
+		fuzzMode = -1
+	}
+	if fuzzMode == -1 {
+		fmt.Printf("Unknown fuzzType %s", *fuzzType)
```
Comment (nitika18) [inline meqa/mqgo/main.go]: Let's print the expected values as well.

File: meqa/mqgo/main.go [hunk 2]
```diff
@@ -311,12 +314,37 @@ func runMeqa(meqaPath, swaggerFile, testPlanFile, resultPath,
 		os.Exit(1)
 	}
 
+	fuzzMode := 0
+	switch strings.ToLower(*fuzzType) {
```
Comment (nitika18) [inline meqa/mqgo/main.go:318]: Does it make sense to have fuzz type as all?

File: meqa/mqplan/dsl.go [hunk 3]
```diff
@@ -30,6 +32,29 @@ import (
...
+
+	StatusSuccess             = "success" // 2XX
+	StatusCodeOk              = 200       // Create success
+	StatusCodeNoResponse      = 204       // Delete success
+	StatusCodeBadRequest      = 400       // Due to incorrect body parameters
+	StatusCodeTooManyRequests = 429       // API rate limiting
+
+	FuzzPositive = 1
+	FuzzDataType = 2
+	FuzzNegative = 3
```
Comment (nitika18) [inline meqa/mqplan/dsl.go:46]: This is repetitive from https://github.com/AdityaVallabh/swagger_meqa/blob/a8e3c9f242cc2e713d2c5bcf83be6f7701f2e948/meqa/mqgo/main.go#L321 . Let's try to move this content to a single place and read from there.

File: meqa/mqplan/plan.go [hunk 4]
```diff
@@ -185,6 +189,14 @@ func (plan *TestPlan) InitFromFile(path string, db *mqswag.DB) error {
 	return nil
 }
 
+func (plan *TestPlan) ReadFails(path string) {
+	data, err := ioutil.ReadFile(filepath.Join(path, MeqaFails))
+	err = yaml.Unmarshal([]byte(data), &plan.Failures)
+	if err != nil {
+		mqutil.Logger.Printf("error: %v", err)
```
Comment (nitika18) [inline meqa/mqplan/plan.go]: We are simply logging error and moving forward. Shouldn't we stop the execution ?

File: meqa/mqswag/parser.go [hunk 5]
```diff
@@ -125,6 +153,105 @@ func GetMeqaTag(desc string) *MeqaTag {
...
+func filter(doneData, allData, dataset *map[string][]interface{}, batchSize int) {
+	doneMap := make(map[string]map[interface{}]bool)
+	for k, v := range *doneData {
+		doneMap[k] = make(map[interface{}]bool)
+		for _, i := range v {
+			doneMap[k][i] = true
+		}
+	}
+	allDone := true
+	for t := 0; t < 2 && allDone; t++ {
```
Comment (nitika18) [inline meqa/mqswag/parser.go:189]: What is `t` ?

File: meqa/mqgo/main.go (hunk 1 above)
Comment (AdityaVallabh) [reply meqa/mqgo/main.go]: Added

File: meqa/mqswag/parser.go (hunk 5 above)
Comment (AdityaVallabh) [reply meqa/mqswag/parser.go:189]: Discussed offline and added comments

**Pull request merged**